src/index.js

```diff
@@ -169,51 +176,60 @@ export default class OasisApp {
   }
 
   async deviceInfo() {
-    return this.transport.send(0xe0, 0x01, 0, 0, Buffer.from([]), [0x9000, 0x6e00]).then((response) => {
-      const errorCodeData = response.slice(-2);
-      const returnCode = errorCodeData[0] * 256 + errorCodeData[1];
+    return this.transport
+      .send(0xe0, 0x01, 0, 0, Buffer.from([]), [0x9000, 0x6e00])
+      .then((response) => {
+        const errorCodeData = response.slice(-2);
+        const returnCode = errorCodeData[0] * 256 + errorCodeData[1];
+
+        if (returnCode === 0x6e00) {
+          // eslint-disable-next-line no-throw-literal
+          throw {
```

Comment:
woa is this consistent with the return-instead-of-throw style in the rest of this repo?

Reply:
Yeah, it gets caught in the next `.then` with `processErrorResponse`. But I see custom error_message will get lost :/

Reply:
hm, how far off is the custom error message from the one supplied by processErrorResponse?

Reply:
`Oasis app does not seem to be open` vs `This command is only available in the Dashboard`

Reply:
oh so I take it 0x6e00 is a generic "you're in the wrong app" code and we have this additional logic to give a hint of what app the user should open

Reply:
... and is it that `error_message` is entirely added from within this library?

Reply:
yeah

Reply:
I have an alternative solution based on the following weirdness.
```ts
// returns { a: number; b?: undefined } | { a: number; b: number }
function a() {
  if (Math.random()) {
    return { a: 5 }
  }
  return { a: 6, b: 7 }
}

// returns { a: number } | { a: number; b: number }
function b() {
  if (Math.random()) {
    const r1 = { a: 5 }
    return r1
  }
  return { a: 6, b: 7 }
}

// returns { a: number; b: number } | { a: number }
function c() {
  if (Math.random()) {
    return { a: 5 }
  }
  const r2 = { a: 6, b: 7 }
  return r2
}

// returns { a: number }
function d() {
  if (Math.random()) {
    const r1 = { a: 5 }
    return r1
  }
  const r2 = { a: 6, b: 7 }
  return r2
}
```

Before PR we had a). I'll use c) and add a type check to cause errors if someone refactors

Reply:
how do (a) and (d) work? weird

Reply:
No idea :shrug: